homeassistant/components/noaa_tides/config_flow.py

```diff
@@ -0,0 +1,145 @@
+)
+from .helpers import get_default_unit_system, get_station_unique_id
+
+DATA_SCHEMA_DICT = {
+    vol.Required(
+        CONF_STATION_ID,
+    ): str,
+}
+
+
+def get_options_schema_dict(hass: HomeAssistant | None = None):
+    """Return options schema dict."""
+    return {
+        vol.Optional(
+            CONF_UNIT_SYSTEM,
+            default=get_default_unit_system(hass),
+        ): SelectSelector(
+            SelectSelectorConfig(
+                {
+                    "options": UNIT_SYSTEMS,
+                    "translation_key": "unit_systems",
+                }
+            )
+        ),
+        vol.Optional(CONF_TIME_ZONE, default=DEFAULT_TIMEZONE): SelectSelector(
+            SelectSelectorConfig(
+                {
+                    "options": TIMEZONES,
+                    "translation_key": "timezones",
+                }
+            )
+        ),
+    }
+
+
+OPTIONS_FLOW = {
+    "init": SchemaFlowFormStep(vol.Schema(get_options_schema_dict())),
+}
+
+
+class NoaaTidesConfigFlow(ConfigFlow, domain=DOMAIN):
+    """NOAA Tides config flow."""
+
+    VERSION = 1
+    MINOR_VERSION = 1
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: ConfigEntry,
+    ) -> SchemaOptionsFlowHandler:
+        """Get the options flow for this handler."""
+        return SchemaOptionsFlowHandler(config_entry, OPTIONS_FLOW)
+
+    async def async_step_import(self, import_data: dict[str, Any]) -> ConfigFlowResult:
+        """Import a config entry from configuration.yaml."""
+        return await self.async_step_user(import_data)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Step when user initializes a integration."""
+
+        input_schema = vol.Schema(DATA_SCHEMA_DICT).extend(
+            get_options_schema_dict(self.hass)
+        )
+
+        if user_input is not None:
+            station_id = user_input[CONF_STATION_ID]
+
+            unique_id = get_station_unique_id(station_id)
+            await self.async_set_unique_id(unique_id)
+            self._abort_if_unique_id_configured()
+
+            unit_system = user_input.get(
+                CONF_UNIT_SYSTEM, get_default_unit_system(self.hass)
+            )
+            timezone = user_input.get(CONF_TIME_ZONE, DEFAULT_TIMEZONE)
+
+            try:
+                station = await self.hass.async_add_executor_job(
+                    Station, station_id, unit_system
+                )
+            except KeyError:
+                _LOGGER.error(
+                    "%s Sensor station_id %s does not exist", NAME, station_id
+                )
+
+                return self.async_show_form(
+                    step_id="user",
+                    data_schema=self.add_suggested_values_to_schema(
+                        input_schema, user_input
+                    ),
+                    errors={"base": "station_not_found"},
+                )
```

Comment:
I would suggest to just fall back to the `self.async_show_form` on line 140. This way you can just have one form call and nothing more

Reply:
That makes sense. I'm not sure the bset way to do this but I gave it a shot. Let me know if there is a better way.